### PR TITLE
Improve slug generation

### LINKED ISSUE #205: Improve slug generation
1. Move slug generation from the web app to an indexing step.
2. Index most languages (en, es, hi, ...) based on max work length, otherwise (cjk, th, ..) use max slug length to truncate.
3. Research has shown that removing stop words can help improve slug quality and SEO. Filter them out.

### PR DESCRIPTION
**Pull request by eliotjordan**
Closes #205

- Moves slug generation to solr indexing.
- Use slug values in solr doc to generate item URL.

### CODE REVIEW

File: solr/conf/solrconfig.xml [hunk 1]
```diff
@@ -781,9 +781,10 @@
     </lst>
   </initParams>
 
+  <!-- <initParams path="/update"> -->
```
Comment (hackartisan) [inline solr/conf/solrconfig.xml:784]: remove this?

File: solr/conf/slug.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  var words = title
+              .replace(/[.,\/#!$%\^&\*'";:{},<=>\-_@`~\[\]()]/g,'')
+              .replace(/  +/g, ' ')
+              .toLowerCase()
+              .split(' ')
+
+  // Filter out stopwords
+  if (lang in stopwords) {
+    filtered_words = words
+                    .filter(function(el) { return stopwords[lang].indexOf(el) == -1 })
```
Comment (hackartisan) [inline solr/conf/slug.js:18]: I'm not sure what `el` means. maybe element? Maybe we could change this to `word`?

File: solr/conf/slug.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                    .filter(function(el) { return el != null; });
+
+    // If number of filtered words is less than three, we won't have a good
+    // slug. Use the unfiltered word list instead.
```
Comment (hackartisan) [inline solr/conf/slug.js:22]: really appreciate the comments in this file!

**Review by hackartisan — CHANGES_REQUESTED** [review]
This looks great to me; I left a couple of small comments.

Does this bring us to title slugs for all languages? Do we have tests for any rtl languages?

File: solr/conf/solrconfig.xml (hunk 1 above)
Comment (eliotjordan) [reply solr/conf/solrconfig.xml]: Done

File: test/dpul_collections/solr_test.exs [hunk 4]
```diff
@@ -135,4 +137,55 @@ defmodule DpulCollections.SolrTest do
...
+    }
+
+    Solr.add([doc], active_collection())
+    Solr.commit(active_collection())
+
+    assert Solr.find_by_id("3cb7627b-defc-401b-9959-42ebc4488f74")["slug_s"] ==
+             "玉機微義-五十卷-徐用誠輯-劉純續輯"
+  end
+
+  test "slug generation with rtl langauge" do
```
Comment (hackartisan) [inline test/dpul_collections/solr_test.exs:179]: 🎉

**Review by hackartisan — APPROVED** [review]

**Pull request merged**